tests/data/test_order.py

```diff
@@ -0,0 +1,172 @@
+    order_filled_sup_side_sell_inst.filled_price = 15.114778
+    order_filled_sup_side_sell_inst.created_last_price = 7.265
+    assert order_filled_sup_side_sell_inst.get_profitability() == (1 - 15.114778 / 7.265)
+
+    # Test filled_price < create_last_price
+    # test side SELL
+    order_filled_sup_side_sell_inst = Order(trader_inst)
+    order_filled_sup_side_sell_inst.side = TradeOrderSide.SELL
+    order_filled_sup_side_sell_inst.filled_price = 11.556877
+    order_filled_sup_side_sell_inst.created_last_price = 20
+    assert order_filled_sup_side_sell_inst.get_profitability() == (1 - 20 / 11.556877)
+
+    # test side BUY
+    order_filled_sup_side_sell_inst = Order(trader_inst)
+    order_filled_sup_side_sell_inst.side = TradeOrderSide.BUY
+    order_filled_sup_side_sell_inst.filled_price = 8
+    order_filled_sup_side_sell_inst.created_last_price = 14.35
+    assert order_filled_sup_side_sell_inst.get_profitability() == (-(1 - 14.35 / 8))
+
+    # Test filled_price == create_last_price
+    # test side SELL
+    order_filled_sup_side_sell_inst = Order(trader_inst)
+    order_filled_sup_side_sell_inst.side = TradeOrderSide.SELL
+    order_filled_sup_side_sell_inst.filled_price = 1517374.4567
+    order_filled_sup_side_sell_inst.created_last_price = 1517374.4567
+    assert order_filled_sup_side_sell_inst.get_profitability() == 0
+
+    # test side BUY
+    order_filled_sup_side_sell_inst = Order(trader_inst)
+    order_filled_sup_side_sell_inst.side = TradeOrderSide.BUY
+    order_filled_sup_side_sell_inst.filled_price = 0.4275587387858527
+    order_filled_sup_side_sell_inst.created_last_price = 0.4275587387858527
+    assert order_filled_sup_side_sell_inst.get_profitability() == 0
+
+
+async def test_check_last_prices(trader_simulator):
+    config, exchange_manager_inst, trader_inst = trader_simulator
+    order_inst = Order(trader_inst)
+
+    # test price in last trades
+    # test inferior TRUE
+    max_price = 10
+    min_price = 4
+    order_inst.creation_time = time.time()
+    recent_trades = [{"price": random_price(min_value=min_price, max_value=max_price), "timestamp": time.time()}
+                     for _ in range(0, SIMULATOR_LAST_PRICES_TO_CHECK)]
+
+    # append validating trade
+    recent_trades.append({"price": min_price, "timestamp": time.time()})
+    assert order_inst.check_last_prices(recent_trades, max_price, inferior=True)
+
+    # test inferior FALSE
+    max_price = 10.454677
+    min_price = 2.4273
+    order_inst.creation_time = time.time()
+    recent_trades = [{"price": random_price(min_value=min_price, max_value=max_price), "timestamp": time.time()}
+                     for _ in range(0, SIMULATOR_LAST_PRICES_TO_CHECK)]
+
+    # append validating trade
+    recent_trades.append({"price": max_price, "timestamp": time.time()})
+    assert order_inst.check_last_prices(recent_trades,
+                                        random_price(min_value=min_price, max_value=max_price - 1),
```

Comment:
won't this make random failures sometimes ?

Reply:
It was already a random generation. I'll improve that